core/common/modules/dev-tools/deprecation.php

```diff
@@ -0,0 +1,169 @@
+<?php
+namespace Elementor\Core\Common\Modules\DevTools;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Deprecation {
+	const SOFT_VERSIONS_COUNT = 4;
+	const HARD_VERSIONS_COUNT = 8;
+
+	/**
+	 * @var Deprecation
+	 */
+	private $deprecation;
+
+	public function __construct() {
+		$this->deprecation = new Deprecation();
+	}
+
+	/**
+	 * Get total of major.
+	 *
+	 * @param object $parsed_version
+	 *
+	 * @return int
+	 */
+	private function get_total_major( $parsed_version ) {
+		$major2 = $parsed_version->major2;
+		$major2 = $major2 > 9 ? 9 : $major2;
+		$minor = 0;
+
+		$total = intval( "{$parsed_version->major1}{$major2}{$minor}" );
+
+		if ( $total > 99 ) {
+			$temp = $total / 10;
+			$total = $temp ? $temp : intval( substr( $total, 0, 2 ) );
+		} else {
+			$total = intval( $total / 10 );
+		}
+
+		return $total;
+	}
+
+	/**
+	 * Get's next version.
+	 *
+	 * @param string $version
+	 * @param int $count
+	 *
+	 * @return string|false
+	 */
+	public function get_next_version( $version, $count = 1 ) {
+		$version = $this->parse_version( $version );
+
+		if ( ! $version ) {
+			return false;
+		}
+
+		$version->total = $this->get_total_major( $version );
+
+		$version->total += $count;
+
+		if ( $version->total > 9 ) {
+			$version->major1 = intval( $version->total / 10 );
+			$version->major2 = intval( $version->total % 10 );
```

Comment:
Why do you need the `intval` call here?

Reply:
since i don't the decimal remainder

Reply:
The result of `$version->total % 10` will always be an integer

Reply:
Use a Yoda condition

Reply:
Handled.